app/assets/javascripts/discourse/app/components/group-smtp-email-settings.js

```diff
@@ -0,0 +1,99 @@
+import Component from "@ember/component";
+import { later } from "@ember/runloop";
+import I18n from "I18n";
+import bootbox from "bootbox";
+import { isEmpty } from "@ember/utils";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { on } from "discourse-common/utils/decorators";
+import { action } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+
+export default Component.extend({
```

Comment:
I thought the same thing but honestly I think they are simple enough to just keep separate; I think it would add more unnecessary indirection to try and split out small parts into reusable components that are then only reused for these two components.

**Having said that**, I will move the provider-specific logic of `prefillSettings` out into a separate reusable function that we can then expand later easily.